### ISSUE TITLE
Display "evaluating" status for nodes

### ISSUE THREAD

**Issue by vitvakatu** [enso-org/enso#7577]
In #7435 we implemented displaying "pending" status for nodes.

We also need some indication for the nodes that a currently evaluated by the engine, as evaluating a single node can take several minutes in some cases.

Can we make the icon of the currently evaluating component into a spinner?

**Comment by JaroslavTulach** [comment]
Preferably this should work in orchestration with progress API

**Comment by farmaazon** [comment]
> Preferably this should work in orchestration with progress API

I see `progress` field in `Pending` update. We could display node as being evaluated if `progress != 0`. If you know better API for this, please let me know.

**Comment by JaroslavTulach** [comment]
> > Preferably this should work in orchestration with progress API
> 
> I see `progress` field in `Pending` update. We could display node as being evaluated if `progress != 0`. If you know better API for this, please let me know.

Yes, that was my idea, when I [introduced the progress field](https://github.com/enso-org/enso/commit/835ac0521810d3bba62f21d2ae512300038efe8a#diff-d9220bac392a7f1b1c979ccf7e815355434e81635bca2af7c5480615cf3b36f2R387). Thus it'd be:
- if there is no `progress` field, the node is _pending and not yet being evaluated_
- if there is some value in `progress` field the node is _pending and being evaluated_

All that's needed is the IDE to use such info and _language server to fill it_:
- non-IDE part tracked as #12009

**Comment by farmaazon** [comment]
Thanks!

**Comment by farmaazon** [comment]
Refinement notes:
* Do not focus on tracking progress, because it's not implemented in the engine anyway. Just check if > 0.
* May be implemented as spinner in place of node's icon, with cross-fade animation, so there will be no flickering.
